AWSutils.sh

```diff
@@ -376,13 +378,51 @@ function InstallSSMagent {
    fi
 }
 
+# Install AWS CFN Bootstrap
+function InstallCfnBootstrap {
+   if [[ -z ${CFNBOOTSTRAP:-} ]]
+   then
+      err_exit "AWS CFN Bootstrap not requested for install. Skipping..." NONE
+   elif [[ ${CFNBOOTSTRAP} == *.tar.gz ]]
+   then
+      local TMPDIR
+      TMPDIR=$(chroot "${CHROOTMNT}" mktemp -d)
+
+      err_exit "Fetching ${CFNBOOTSTRAP}..." NONE
+      curl -sL "${CFNBOOTSTRAP}" -o "${CHROOTMNT}${TMPDIR}/aws-cfn-bootstrap.tar.gz"
+
+      err_exit "Installing AWS CFN Bootstrap..." NONE
+      chroot "${CHROOTMNT}" python3 -m pip install "${TMPDIR}/aws-cfn-bootstrap.tar.gz" || \
```

Comment:
If you want to collapse things, at all, you might be able to do this with:

~~~
python3 -m pip install <( curl -sL "${CFNBOOTSTRAP}" )
~~~

Reply:
Even more compact, pip supports urls, so this works: `python3 -m pip install "${CFNBOOTSTRAP}"`

But I didn't do that, because pip doesn't extract the cfn-hup service init file that way. So I'm pulling it out using `tar` a little later in the function. So curl to download once, then I'm using the local file twice.